### PR TITLE
Add publishing for .NET 8 channels

### PR DESCRIPTION
**Pull request by mmitche**
### To double check:

* [ ] The right tests are in and and the right validation has happened.  Guidance:  https://github.com/dotnet/arcade/tree/main/Documentation/Validation

### CODE REVIEW

**Comment by mmitche** [comment]
I've updated the PR to add back the public RC 1 channels, which I should not have removed.

File: src/Microsoft.DotNet.Build.Tasks.Feed/src/model/PublishingConstants.cs [hunk 1]
```diff
@@ -373,6 +404,46 @@ public enum BuildQuality
...
+                false,
+                PublishingInfraVersion.All,
+                new List<string>() { "7.0.1xx-rc1", "7.0-rc1" },
+                DotNet7Feeds,
+                PublicAndInternalSymbolTargets,
+                filenamesToExclude: FilenamesToExclude),
+
+            // ".NET 7 RC 2",
+            new TargetChannelConfig(
+                2845,
```
Comment (riarenas) [inline src/Microsoft.DotNet.Build.Tasks.Feed/src/model/PublishingConstants.cs:429]:
```suggestion
                3078,
```

File: src/Microsoft.DotNet.Build.Tasks.Feed/src/model/PublishingConstants.cs [hunk 2]
```diff
@@ -373,6 +404,46 @@ public enum BuildQuality
...
+                false,
+                PublishingInfraVersion.All,
+                new List<string>() { "7.0-rc2" },
+                DotNet7Feeds,
+                PublicAndInternalSymbolTargets,
+                filenamesToExclude: FilenamesToExclude),
+
+            // ".NET 7 SDK 7.0.1xx RC 2",
+            new TargetChannelConfig(
+                3078,
```
Comment (riarenas) [inline src/Microsoft.DotNet.Build.Tasks.Feed/src/model/PublishingConstants.cs:439]:
```suggestion
                3076,
```

**Review by riarenas — COMMENTED** [review]
I think 2 ids don't match what I get from darc and left suggestions for what I think are the correct values. Otherwise lgtm

**Comment by mmitche** [comment]
Good eye, thanks.

**Review by riarenas — APPROVED** [review]

**Review by epananth — APPROVED** [review]

**Pull request merged**